### PR TITLE
8293020: jmod should not be treated as "small" tool for large modules

### PR DESCRIPTION
**Pull request by shipilev**
This is similar to [JDK-8245168](https://bugs.openjdk.org/browse/JDK-8245168), but the blanket change to allow all modules be compiled with default options is a net loss. Instead, we can hand-pick the major offenders (large modules) where running jmod with normal tool options improves performance.

I instrumented the jmod to tell me the times it needs to create individual modules, and hand-picked three top modules that take multiple seconds to run.

Motivational `make clean-images images` times:

```
# x86_64 Server, release

# Baseline
real	0m12.040s
user	1m4.872s
sys	0m10.805s

# Patched
real	0m10.785s  ; <--- 1.2s faster
user	1m7.031s
sys	0m10.985s

# x86_64 Server, fastdebug

# Baseline
real	0m19.263s
user	2m42.317s
sys	0m18.537s

# Patched
real	0m17.911s ;  <--- 1.1s faster
user	2m52.810s
sys	0m19.092s


# x86_64 Server, slowdebug

# Baseline
real	0m44.799s
user	10m7.106s
sys	0m17.578s

# Patched
real	0m46.975s   ; <--- 2.5 sec slower
user	11m1.155s
sys	0m17.060s
```

I think we can accept the `slowdebug` regression in favor of improvements on `release` and `fastdebug` that most people seem to be building every day.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8293020](https://bugs.openjdk.org/browse/JDK-8293020): jmod should not be treated as "small" tool for large modules

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)
 * [Magnus Ihse Bursie](https://openjdk.org/census#ihse) (@magicus - **Reviewer**)
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10062/head:pull/10062` \
`$ git checkout pull/10062`

Update a local copy of the PR: \
`$ git checkout pull/10062` \
`$ git pull https://git.openjdk.org/jdk pull/10062/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10062`

View PR using the GUI difftool: \
`$ git pr show -t 10062`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10062.diff">https://git.openjdk.org/jdk/pull/10062.diff</a>

</details>

### CODE REVIEW

File: make/CreateJmods.gmk [hunk 1]
```diff
@@ -226,6 +226,12 @@ else
   JMOD_FLAGS += --exclude '**{_the.*,_*.marker*,*.diz,*.debuginfo,*.dSYM/**,*.dSYM}'
 endif
 
+# Unless we are creating a very large module, use the small tool JVM options
+JMOD_SMALL_FLAGS=
+ifeq ($(findstring $(MODULE), java.base java.desktop jdk.localedata), )
+    JMOD_SMALL_FLAGS += $(JAVA_TOOL_FLAGS_SMALL)
```
Comment (erikj79) [inline make/CreateJmods.gmk:232]: Two space indent please, otherwise looks good.

**Review by erikj79 — APPROVED** [review]

File: make/CreateJmods.gmk [hunk 2]
```diff
@@ -226,6 +226,12 @@ else
   JMOD_FLAGS += --exclude '**{_the.*,_*.marker*,*.diz,*.debuginfo,*.dSYM/**,*.dSYM}'
 endif
 
+# Unless we are creating a very large module, use the small tool JVM options
+JMOD_SMALL_FLAGS=
```
Comment (erikj79) [inline make/CreateJmods.gmk:230]: Missed this at first, but we try to stick to := assignment unless we really need the lazy evaluation. Not really an issue here other than consistency.

**Review by erikj79 — APPROVED** [review]

File: make/CreateJmods.gmk (hunk 2 above)
Comment (shipilev) [reply make/CreateJmods.gmk:230]: Done in new commit.

**Review by erikj79 — APPROVED** [review]

**Review by magicus — APPROVED** [review]

**Comment by shipilev** [comment]
If anyone from core-libs crowd wants to chime in, welcome. Otherwise, I'll integrate soon.

**Review by AlanBateman — APPROVED** [review]

**Pull request closed without merging**